components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/client/PubSubApiClient.java

```diff
@@ -407,10 +350,22 @@ public void onError(Throwable throwable) {
                             LOG.debug("logged in {}", consumer.getTopic());
                         }
                         case PUBSUB_ERROR_CORRUPTED_REPLAY_ID -> {
-                            LOG.error("replay id: " + replayId
-                                      + " is corrupt. Trying to recover by resubscribing with LATEST replay preset");
-                            replayId = null;
-                            initialReplayPreset = ReplayPreset.LATEST;
+                            String currReplayId = null;
+                            if (replayId != null) {
+                                currReplayId = replayId;
+                            } else if (initialReplayPreset == ReplayPreset.CUSTOM) {
+                                currReplayId = initialReplayId;
+                            }
+                            if (fallbackToLatestReplayId) {
+                                initialReplayPreset = ReplayPreset.LATEST;
+                                LOG.error("replay id: " + currReplayId
+                                          + " is corrupt. Trying to recover by resubscribing with LATEST replay preset");
+                                replayId = null;
+                            } else {
+                                consumer.getExceptionHandler().handleException(new InvalidReplayIdException(
+                                        "Invalid replay id: " + currReplayId,
```

Comment:
Maybe use the same word as above, that its corrupt, sine the error case is  ERROR_CORRPUTED